### PR TITLE
update legacy facts under /templates

### PR DESCRIPTION
**Pull request by rrotter**
See:
* https://www.puppet.com/docs/puppet/8/upgrading-from-puppet7-to-puppet8.html#legacy-facts-deprecation
* #758

### CODE REVIEW

File: templates/profile/kubernetes/dns/hosts_04_ipv4_hostname.erb
Comment (rrotter) [inline templates/profile/kubernetes/dns/hosts_04_ipv4_hostname.erb:1]: This is one that I fixed in the previous pass. Using `@networking` for consistancy, `@networking`  and `@facts['networking']` should be equivalent.

**Review by daaang — APPROVED** [review]
This all looks sane. Are you certain that `@networking` is accessible from all these templates? I don't have any certainty around whether each corresponding puppet code will need

```puppet
$networking = $::networking
```

This is probably at least *somewhat* covered by the tests...

**Comment by rrotter** [comment]
> This is probably at least _somewhat_ covered by the tests...

In puppet code `$::facts['networking'] == $::networking` (where I've used the latter universally). I think this maps to erb templates such that `@facts['networking'] == @networking`. If this is wrong I'd just use `@facts['networking']` in the templates. I'd expect to see a lot of failing tests in that case, but I'm also going to test this out in an environment with a few hosts before merging.

**Comment by rrotter** [comment]
Converted to draft as this needs to be tested further. Not to be merged before Monday.

**Comment by rrotter** [comment]
There were a few places where `$ipaddress`, _et al._ were local variables so `@ipaddress` in the template was actually correct as we need the local variable and not the fact. I've reverted the cases of this issue that I found.

**Pull request merged**